yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -1160,6 +1123,216 @@ describe('AVM simulator: transpiled Noir contracts', () => {
       });
     });
   });
+  describe('Side effects including merkle checks', () => {
```

Comment:
New test group that uses real merkle trees and checks that traced merkle preimages/paths are right.